.travis.yml

```diff
@@ -20,20 +20,29 @@ matrix:
             - sourceline: "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-11 main"
               key_url: "https://apt.llvm.org/llvm-snapshot.gpg.key"
           packages:
+            - clang-11
             - faketime
-            - protobuf-compiler
             - libsgx-dcap-ql-dev
-            - clang-11
             - musl-tools
+            - protobuf-compiler
       rust:
         - stable
       env:
         - RUST_BACKTRACE=1
         - CFLAGS_x86_64_fortanix_unknown_sgx="-isystem/usr/include/x86_64-linux-gnu -mlvi-hardening -mllvm -x86-experimental-lvi-inline-asm-hardening"
         - CC_x86_64_fortanix_unknown_sgx=clang-11
       before_script:
+        # Travis-CI supplies clang version 16 by itself, in /usr/local
+        # However, our current build fails if llvm-16 is also available in the system. This is likely a bug,
+        # created and reported here: https://github.com/fortanix/rust-sgx/issues/465
+        # Once this bug is solved, please feel free to remove this workaround.
+        #
+        # Note that this clang-16 is some Travis-CI magic and not a standard Debian/Ubuntu Apt package.
+        # We're thus forced to force-remove the underlying directory in /usr/local
+        - ls -aFl /usr/local | grep clang
+        - sudo rm -rf /usr/local/*clang*
         - rustup target add x86_64-fortanix-unknown-sgx x86_64-unknown-linux-musl
-        - rustup toolchain add nightly
+        - rustup toolchain add --profile minimal nightly
```

Comment:
This doesn't seem required.

Reply:
Yes, it's unrelated. Should I move it to a separate PR? I think it's just nicer to always specify minimal dependencies/profiles/downloads of any kind for CI

Reply:
Oh can you also revert this change? I'm not against it, but it should be a separate PR then. PRs are preferably the minimal set of changes to fix an issue / add a new feature. I was considering a similar comment about your changes to the `package` key, but you can stretch it a bit and say you're doing changes to the installed packages anyway.

Reply:
If you mean the `packages` section, then I've just ordered them alphabetically. (Ordered things are just nicer.)

My personal preferences would be to be practical and merge as-is (especially since we don't have release trains and are asked to rebase if `master` changes). But that's my personal preference, I can easily go with what you suggest

Should I split or revert?

Reply:
@jethrogb @raoulstrackx --^

Reply:
Personally I would also only rearrange the packages when I'm adding one.
We don't have an agreement on whether its still acceptable, so let's be puristic and only keep the minimal change required: `sudo rm -rf /usr/local/*clang*` and its comment.

Reply:
Sure! To maybe explain a bit more on my personal take on this, I prefer the "development boy scout rule":
> The boy scout rule is about making small, continuous improvements to your code. If you are fixing a bug, and find the code around it is a mess, fix a bit of that mess. You don’t have to make it perfect, you don’t have to fix it all, but improve it just a little bit.

https://backendtea.com/post/boyscout-rule/

So personally, I include the "area around the main task" as a possible scope of changes, but I exclude "unrelated" changes, e.g. something in "another area".

This view also has downsides, it's still a trade off. For example, if you'd have to revert said branch, you'd also revert the "cleaning up" part around. On the other hand, strict separation leads to no "cleaning up" at all, because most people wouldn't really submit a separate PR that fixes some typos or adds some links to the documentation of a method that was read and used but not changed in the PR.

That all being said, I'll remove everything but the minimal code changes to address the issue now

Reply:
Done :checkered_flag: